app/controllers/uploads_controller.rb

```diff
@@ -12,22 +10,29 @@ def new
     @upload = new_upload(params[:csv_type])
     return if @upload.csv_type_check?
 
-    log_and_display_error(errors_for_alert([@upload]), 'Warning')
+    alert_and_log(@upload.errors.full_messages.join(', '))
```

Comment:
Actually, the whole check is superfluous if we say that the only legitimate way to reach this action is via a link. Then there is no point in checking whether the argument is valid. However, if we do not say this (e.g., someone manually enters the url in a browser) then it needs to be checked.